Bottom: 7b14bf4de6e1cc019909ac9ddea4bb25ec59fbea
Top:    090a807e09a659a09d49a7c5105ec0c95eaddffd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-18 23:18:58 +0100

hid/gtk: Don't call CrosshairOn() when re-entering the viewport

The GTK HID doesn't turn the crosshair off, so we don't turn it back on
if we default to having it switched on in the first place. Make that
default change, and remove the CrosshairOn() call.

(Also tested not to break the Lesstif HID).


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 7df5299..eb19257 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1124,7 +1124,7 @@ InitCrosshair (void)
   /* fake a crosshair off entry on stack */
   CrosshairStackLocation = 0;
   CrosshairStack[CrosshairStackLocation++] = true;
-  Crosshair.On = false;
+  Crosshair.On = true;
 
   /* set initial shape */
   Crosshair.shape = Basic_Crosshair_Shape;
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 331f214..268988d 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -728,7 +728,6 @@ ghid_port_window_enter_cb (GtkWidget * widget,
     {
       ghid_screen_update ();
     }
-  CrosshairOn ();
   return FALSE;
 }
